rebound/simulation.py

```diff
@@ -1186,7 +1191,10 @@ def additional_forces(self):
         raise AttributeError("You can only set C function pointers from python.")
     @additional_forces.setter
     def additional_forces(self, func):
-        self._afp = AFF(func)
+        if isinstance(func, _CFuncPtr):
```

Comment:
There used to be an issue with retaining a reference to the callback function type. I don't remember all the details. I don't know if this change will resurface the issue.